packages/block-editor/src/components/block-list/use-block-props/use-zoom-out-mode-exit.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useRefEffect } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+import { unlock } from '../../../lock-unlock';
+
+/**
+ * Allows Zoom Out mode to be exited by double clicking in the selected block.
+ *
+ * @param {string} clientId Block client ID.
+ */
+export function useZoomOutModeExit( { editorMode } ) {
+	const { __unstableSetEditorMode } = unlock(
+		useDispatch( blockEditorStore )
+	);
+
+	return useRefEffect(
+		( node ) => {
+			if ( editorMode !== 'zoom-out' ) {
```

Comment:
Instead of having the editorMode as a dependency of the useRefEffect and regenerating a new ref / new event listener everytime this change. We could just call getEditorMode within onDoubleClick event listener (Like we do for useNavModeExit

This would also remove the need to pass the editorMode top/down through the context...

Reply:
I'm seeing the opposite comment above. I think personally the performance impact of having the double click always there is less than re-rendering when the editorMode changes, it's also way simpler in terms of code. But I'm ok if you all disagree.

Reply:
I had the same thought process. I believe @jeryj did some performance profiling on this so he may be able to answer more concretely.

Reply:
I had done profiling on the page and not adding the event listener used a lot less resources. The metrics for typing are going up. I'm having a hard time isolating a PR that is causing it. It's odd to me that adding the mode to the block props would cause that, as the editing mode doesn't really change often and definitely isn't changed in the middle of the performance tests. I can make another branch that reworks this to this older style and we can compare.

Reply:
https://github.com/WordPress/gutenberg/pull/65326

@getdave the original way we did this also included passing `clientId` to the `refEffect` which would have made a difference in the profiling. We don't need that anymore, so hopefully the performance will be better now.

Reply:
I don't see any major changes [in performance with either approach](https://github.com/WordPress/gutenberg/actions/runs/10851356209/job/30114916480?pr=65326).

Reply:
The typing metic was probably affected by a new store subscription. See #65337.